variables.tf

```diff
@@ -4,36 +4,73 @@
 # You must provide a value for each of these parameters.
 # ------------------------------------------------------------------------------
 
-variable "subnet_id" {
-  type        = string
-  description = "The ID of the AWS subnet to deploy into (e.g. subnet-0123456789abcdef0)"
-}
-
 # ------------------------------------------------------------------------------
 # OPTIONAL PARAMETERS
 #
 # These parameters have reasonable defaults.
 # ------------------------------------------------------------------------------
-variable "ami_owner_account_id" {
+variable "assessment_images_bucket_base_name" {
+  type        = string
+  description = "The base name to use for the assessment images S3 buckets."
+  default     = "cisa-cool-assessment-images"
+}
+
+variable "assessmentimagesbucketfullaccess_role_description" {
   type        = string
-  description = "The ID of the AWS account that owns the Example AMI, or \"self\" if the AMI is owned by the same account as the provisioner."
-  default     = "self"
+  description = "The description to associate with the IAM role and attached policy that allows full access to the assessment images S3 bucket."
+  default     = "Allows full access to the S3 bucket where assessment images are stored."
 }
 
-variable "aws_availability_zone" {
+variable "assessmentimagesbucketfullaccess_role_name" {
   type        = string
-  description = "The AWS availability zone to deploy into (e.g. a, b, c, etc.)"
-  default     = "a"
+  description = "The name to associate with the IAM role and attached policy that allows full access to the assessment images S3 bucket."
+  default     = "AssessmentImagesBucketFullAccess"
+}
+
+variable "assume_assessmentimagesbucketfullaccess_roles_policy_description" {
+  type        = string
+  description = "The description to associate with the IAM policy that allows a user to assume the IAM roles that allow access to the assessment images S3 bucket in the Images (Production) and Images (Staging) accounts."
+  default     = "Allows assumption of the roles in the Images (Production) and Images (Staging) accounts that allow full access to the assessment images S3 bucket."
 }
 
 variable "aws_region" {
   type        = string
-  description = "The AWS region to deploy into (e.g. us-east-1)"
+  description = "The AWS region to use for the account provisioners (e.g. \"us-east-1\")."
   default     = "us-east-1"
 }
 
+variable "provisionaccount_role_name" {
+  type        = string
+  description = "The name of the IAM role that allows sufficient permissions to provision all AWS resources in the account."
```

Comment:
All of the COOL related repositories I looked at that used a `provisionaccount_role_name` variable so I was following suite. Do we use the remote state instead of a configurable variable in another project?

Reply:
@mcdonnnj, I've been following suit too but I think that method came into being before we started working with remote states.

Reply:
My feeling is that whenever we can replace a variable with something from a remote state, it's worthwhile to do so.  Especially if it's a "core" remote state like `images_production` and `images_staging` here.

Reply:
This is a sensible argument and I made the change in 0edd170.

Reply:
LGTM!